brave/brave-core
# Wayback machine page action

Resolves https://github.com/brave/brave-browser/issues/35820

[simplescreenrecorder-2024-03-25_18.47.48.webm](https://github.com/brave/brave-core/assets/6786187/4fd6d6cf-2c9f-4146-9e01-ad4afeac3bbf)

NOTE: button style is different with design. When we update common dialog's button style, this one will be updated together.

Page icon is shown when page is mssing
![image](https://github.com/brave/brave-core/assets/6786187/8c9bbe43-ee5d-4a5c-8c27-cc9941cc43e8)
![image](https://github.com/brave/brave-core/assets/6786187/993ba621-8c69-4d41-bfd7-44a8ff9d8279)
Loading indicator while checking the wayback url
![image](https://github.com/brave/brave-core/assets/6786187/a6acb5f3-1532-4a5e-8eec-c84090b8f6c2)
Checked badge when wayback url is found and loaded
![image](https://github.com/brave/brave-core/assets/6786187/312f10a8-4b2f-4266-bbc8-8537f39ee58e)

Warning badge when wayback url is not available
![image](https://github.com/brave/brave-core/assets/6786187/e6cb6857-8628-49e1-9163-975d9f18f8a7)
![image](https://github.com/brave/brave-core/assets/6786187/bce4f50d-fb99-46d4-9920-6e2c3eaeb7c1)

cc @diracdeltas 



## Submitter Checklist:

  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`

## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on

## Test Plan:

1. Launch brave
2. Load brave.com/bo and check wayback machine page icon is shown in omnibox
3. Create active NTP and check wayback machine page icon is hidden
4. Activate brave.com/bo tab and click icon
5. Check wayback bubble is launched and click `Check` button
6. Bubble is disappeared and icon has loading indicator till wayback url loads start
7. After loading indicator, check icon has check mark.
8. After waybacl url is loaded click reloading toolbar button and check wayback icon is disappeared
9. Load brave.com/bos and check page icon is shown in omnibox
10. Click the icon and check bubble is launched
11. Click 'Check` button and icon has warning badge
12. Click toolbar back button and check icon doesn't have any badge
13. Toggle off settings from brave://settings/braveHelpTips
14. Load brave.com/bo and check page icon is not shown
15. Toggle on settings from brave://settings/braveHelpTips
16. Load brave.com/bo and check page icon is shown
17. Toggle off settings from brave://settings/braveHelpTips
18. Check tab that loaded brave.com/bo doesn't have page icon anymore
